api/envoy/extensions/filters/http/credential_injector/v3/credential_injector.proto

```diff
@@ -17,26 +17,25 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 option (xds.annotations.v3.file_status).work_in_progress = true;
 
 // [#protodoc-title: Credential Injector]
-// [#not-implemented-hide:]
 // Credential Injector :ref:`configuration overview <config_http_filters_credential_injector>`.
 // [#extension: envoy.filters.http.credential_injector]
 
 // Credential Injector injects credentials into outgoing HTTP requests. The filter configuration is used to retrieve the credentials, or
 // they can be requested through the OAuth2 client credential grant. The credentials obtained are then injected into the Authorization header
 // of the proxied HTTP requests, utilizing either the Basic or Bearer scheme.
 //
-// If the credential is not present, the request will fail with 401 Unauthorized if fail_if_not_present is set to true.
+// If the credential is not present or failed to inject the credential, the request will fail with ``401 Unauthorized`` unless
+// ``allow_request_without_credential`` is set to ``true``.
```

Comment:
Are there typical cases where an injection to a request may fail? If so, consider adding a short comment with these examples.
Alternatively this can be added to the docs, and a link in the proto to the doc can be added.

Reply:
It's unlikely for the general credential type to fail since the credentials are specified in the config. Would you mind adding the example later when the credential type for [Client Credentials Grant](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4) is added? Either my colleague or I will work on that after this PR is completed.